### ISSUE TITLE
Align GDL's nomenclature to Torchgeo/Pytorch Lightning

### ISSUE THREAD

**Issue by remtav** [NRCan/geo-deep-learning#300]
[Torchgeo's glossary](https://torchgeo.readthedocs.io/en/latest/user/glossary.html)

Some examples:
- sample -> chip/patch
- Use "aoi" and "tile" in documentation:
   - aoi: Synonym for [region of interest (ROI)](https://torchgeo.readthedocs.io/en/latest/user/glossary.html#term-region-of-interest-ROI). A particular spatial area to focus on.
   - tile: A single image file taken by a remote sensor like a satellite.

GDL's mode should also be revised. I'd suggest changing all modes to be verbs at the infinitive form:
- sampling -> tile
- train
- inference -> predict
- postprocess
- validate
- ...

**Comment by remtav** [comment]
For training, our SegmentationDataset returns a [dict with "sat_img" and "map_img" as keys](https://github.com/NRCan/geo-deep-learning/blob/develop/utils/create_dataset.py#L121).

Torchgeo uses "image" and "mask" as equivalents to these keys. We should stick to this new standard, also because some torchgeo utilities that could become useful to us expect these keys specifically:

Examples of torchgeo datasets using these keys:
- [Chesapeake](https://github.com/microsoft/torchgeo/blob/76139f60df21644c1498e39ac6a8be508115638c/torchgeo/datasets/chesapeake.py#L588)
- [LandcoverAI](https://github.com/microsoft/torchgeo/blob/6f9589dc077839847a2911fb5813eaf908200e17/torchgeo/datasets/landcoverai.py#L124)
- [Potsdam](https://github.com/microsoft/torchgeo/blob/76139f60df21644c1498e39ac6a8be508115638c/torchgeo/datasets/potsdam.py#L163)

A torchgeo utility expecting these keys specifically:
- [AugmentationSequential](https://github.com/microsoft/torchgeo/blob/6f9589dc077839847a2911fb5813eaf908200e17/torchgeo/transforms/transforms.py#L18)
